qiskit/transpiler/passes/optimize_phase_shift_gates.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Pass for merging/canceling phase-shift gates like u1(theta), T-gates, S-gates and Z-gates
+"""
+import numpy as np
+import qiskit
+from qiskit import QuantumCircuit, ClassicalRegister, QuantumRegister
+from qiskit.transpiler.basepasses import TransformationPass
+import copy
+import sys
+np.set_printoptions(threshold=sys.maxsize)
+import warnings
+warnings.filterwarnings('ignore')
+
+class OptimizePhaseShiftGates(TransformationPass):
+    """merge/cancel phase-shift gates in dag."""
+
+    def run(self, dag):
+        """
+        Run one pass of phase-shift gate optimization and cx cancellation
+        on the circuit. The function looks for the circuit configuration
+        with the lowest number of gates and circuit depth
+
+        Args:
+            dag (DAGCircuit): the directed acyclic graph to run on.
+        Returns:
+            DAGCircuit: Transformed DAG.
+        """
+
+        n = dag.width()  # number of qubits in the circuit
+        T_counter = {}  # Counts the cumulative phase of T-gates and other z-rotation gates
+        T_position_counter = {}  # Keeps track of the locations of T-gates and other z-rot. gates
+        state_tracker = np.zeros((dag.width(), dag.size())).astype('int')
+        for i in range(dag.width()):
+            state_tracker[i][i] = 1 # initial state
+        k = dag.width()  # number of initial variables
+        for e in dag.topological_nodes():
+            if e.data_dict['type'] == 'op':
```

Comment:
This can be simplified to `e.type == 'op'`. There are a few places here where you are accessing the `data_dict` and this should not be necessary, you should be able to simply use the dot notation instead.

Reply:
Thank you! I changed it, and the code runs fine without accessing `data_dict ` at any place :)